frontend/src/components/Follow/Follow.jsx

```diff
@@ -41,10 +59,28 @@ export default function Follow({ ticker, stockData, setStockData }) {
     }
 
     return (
-        <div className="follow">
-            <button onClick={handleFollow}>
-                {stockData.following ? "Following" : "Follow"}
-            </button>
-        </div>
+        <SplitButton
+            title={stockData.following ? "Following" : "Follow"}
+            variant="primary"
+            onClick={handleFollow}
+            className="me-2"
+        >
+            <Dropdown.Item>
+                {follows.length} users follow this stock.
+            </Dropdown.Item>
+            <Dropdown.Divider />
+            {follows?.map((follow) => {
+                return (
+                    <Dropdown.Item
+                        key={follow.User.username}
+                        onClick={() =>
+                            navigate(`/profile/${follow.User.username}`)
+                        }
+                    >
+                        {follow.User.username}
+                    </Dropdown.Item>
+                );
+            })}
```

Comment:
Design thought -- `Follow`, `Like`, and `Dislike` all have a really similar structure with a `SplitButton` and dropdown. Can you think of how you can create a new component that takes in some props so you can reuse that component? 

Main advantages are reducing code duplication and if you want to change something overall structure, avoiding needing to make the changes in all 3 components

Reply:
I'm having trouble imagining how to do it since so many variable names are different.

Reply:
It can be done but I would have to use a ton of if statements right?